docs/DeveloperGuide.md

```diff
@@ -1,38 +1,228 @@
+### Sequence Explanation
+* Upon creation of the `Storage` object, the constructor first checks whether the directory and file
+`${CWD}/data/duke_data.txt` exists, and creates them otherwise.
+* Before entering the main loop, `Duke` will first call `Storage::loadDataExpenses()`, which reads from `duke_data.txt`
+and initializes and *returns* a `ExpenseManager` object containing the budget, `Expenses` and `FutureExpenses` from the data file.
+* Duke then uses the returned `ExpenseManager` object for the current session
+* After every command, `Duke` will then call `Storage::saveExpenses(ExpenseManager)`, which serializes the current state of
+the `ExpenseManager` and writes the serialized data into `duke_data.txt`, making sure that the latest state of the app is always saved.
+
+## Implementation
 
 ## Product scope
 ### Target user profile
 
-{Describe the target user profile}
+Our target user profile are people who want to be financially responsible and want to keep a track of their spending to make informed financial decisions. The users include students, small business owners etc. who are interested in managing their finances efficiently.
 
 ### Value proposition
 
-{Describe the value proposition: what problem does it solve?}
+Money master provides a fast and streamlined way for users to manage their finances. It helps the user manage their finances quickly and efficiently by providing a convenient way to track their daily expenses. It allows the user to create and track their budgets to prevent overspending and adjusting their expenses accordingly.
 
 ## User Stories
 
-|Version| As a ... | I want to ... | So that I can ...|
-|--------|----------|---------------|------------------|
-|v1.0|new user|see usage instructions|refer to them when I forget how to use the application|
-|v2.0|user|find a to-do item by name|locate a to-do without having to go through the entire list|
+| Version | As a ...         | I want to ...                                       | So that I can ...                                                                     |
+|---------|------------------|-----------------------------------------------------|---------------------------------------------------------------------------------------|
+| v1.0    | budgeted student | set a budget limit from which expenses are deducted | track my expenses and avoid overrunning my budget                                     |
+| v1.0    | new user         | add expenses                                        | plan my my expenditure based on the remaining budget                                  |
+| v1.0    | forgetful user   | view my expenses                                    | remove incorrect or outdated entries from my transaction history                      |
+| v1.0    | careless user    | edit my expenses                                    | modify or correct the entries in my transaction history                               |
```

Comment:
Consider adding more specificity to some of your user stories. For example, not only careless users will need to edit expenses, as expenses may need to be edited for a variety of reasons.